modules/arrow/src/geoarrow/convert-geoarrow-to-binary-geometry.ts

```diff
@@ -99,33 +111,50 @@ export function getBinaryGeometriesFromArrow(
         ...BINARY_GEOMETRY_TEMPLATE,
         ...(featureTypes.polygon ? binaryContent : {}),
         polygonIndices: {
-          // TODO why deck.gl's tessellatePolygon performance is not good when using geometryIndicies
-          // even when there is no hole in any polygon
+          // use geomOffset as polygonIndices same as primitivePolygonIndices since we are using earcut to get triangule indices
           value: featureTypes.polygon ? geomOffset : new Uint16Array(0),
           size: 1
         },
         primitivePolygonIndices: {
           value: featureTypes.polygon ? geomOffset : new Uint16Array(0),
           size: 1
-        }
+        },
+        ...(triangles ? {triangles: {value: triangles, size: 1}} : {})
       }
     });
 
     bounds = updateBoundsFromGeoArrowSamples(flatCoordinateArray, nDim, bounds);
+    if (meanCenters) meanCenterArray.push(meanCenters);
   });
 
-  return {binaryGeometries, bounds, featureTypes};
+  // get mean centers
+  const globalMeanCenters: number[][] = [];
```

Comment:
Nit: This could at least be a helper function to make this megafunction a little shorter??